tasks/promote.yml

```diff
@@ -1,63 +1,88 @@
 ---
-- name: Check if image suffix is provided
-  fail:
-    msg: "os_images_name_suffix is empty please provide it."
-  when: os_images_name_suffix is defined and os_images_name_suffix | length == 0
-
-- name: Gather retire candidates info
+- name: Gather candidates info
   openstack.cloud.image_info:
     auth_type: "{{ os_images_auth_type }}"
     auth: "{{ os_images_auth }}"
     cacert: "{{ os_images_cacert | default(omit) }}"
     interface: "{{ os_images_interface | default(omit, true) }}"
-  register: retire_list
-  when: item.rename_image | default(os_images_promote) | bool
+  register: retire_or_promote_list
+  when: >
+       (os_images_list | 
+        selectattr('rename_image', 'defined') |
+        selectattr('rename_image', 'equalto', true) |
+        list | length > 0)
+        or (os_images_promote | bool)
 
 - name: Ensure images for retirement exist
   assert:
     that:
-      - item.name in retire_list.openstack_images | map(attribute='name') | list
+      - promotion_name in retire_or_promote.openstack_images | map(attribute='name') | list
     fail_msg: "The image {{ item.name[:-(os_images_name_suffix | length | int)] }} does not exist."
+  vars:
+    promotion_name: "{{ item.name[:-(os_images_name_suffix | length | int)] }}"
   loop: "{{ os_images_list | list }}"
   when: item.rename_image | default(os_images_promote) | bool
 
-- name: Gather promote candidates info
-  openstack.cloud.image_info:
-    auth_type: "{{ os_images_auth_type }}"
-    auth: "{{ os_images_auth }}"
-    cacert: "{{ os_images_cacert | default(omit) }}"
-    interface: "{{ os_images_interface | default(omit, true) }}"
-  register: promote_list
-  when: item.rename_image | default(os_images_promote) | bool
-
 - name: Ensure images for promotion exist
   assert:
     that:
-      - item.name in promote_list.openstack_images | map(attribute='name') | list
+      - item.name in retire_or_promote_list.openstack_images | map(attribute='name') | list
     fail_msg: "The image {{ item.name }} does not exist."
   loop: "{{ os_images_list | list }}"
   when: item.rename_image | default(os_images_promote) | bool
 
+- name: Set community images
+  command: "{{ os_images_venv }}/bin/openstack image set --community {{ item.name }}"
+  loop: "{{ os_images_list | list }}"
+  when: item.visiblility == 'community'
```

Comment:
@markgoddard this is where the new functionality comes in. The openstack.cloud.image module only lets you set public or private on image creation, so community is set afterwards.

Reply:
Perhaps it depends on the version of the module, but it does seem to support visibility: https://docs.ansible.com/ansible/latest/collections/openstack/cloud/image_module.html

Reply:
@markgoddard ah, i see, I must've been looking at an old version. Updated now